src/features/sanctions/hooks/useChainalysisSanctioned.ts

```diff
@@ -0,0 +1,43 @@
+import { isAddress } from 'viem';
+import { useContractRead } from 'wagmi';
+
+import { useEvmAccount } from '../../wallet/hooks/evm';
+
+// https://go.chainalysis.com/chainalysis-oracle-docs.html
+const ORACLE_ABI = [
+  {
+    inputs: [
+      {
+        internalType: 'address',
+        name: 'addr',
+        type: 'address',
+      },
+    ],
+    name: 'isSanctioned',
+    outputs: [
+      {
+        internalType: 'bool',
+        name: '',
+        type: 'bool',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+] as const;
+const ORACLE_ADDRESS = '0x40C57923924B5c5c5455c48D93317139ADDaC8fb';
+
+export const useChainalysisSanctioned = () => {
+  const evmAddress = useEvmAccount().addresses[0]?.address;
+
+  const sanctioned = useContractRead({
+    abi: ORACLE_ABI,
+    functionName: 'isSanctioned',
+    args: [isAddress(evmAddress) ? evmAddress : '0x'],
+    chainId: 1,
```

Comment:
Can you confirm it's sufficient to check only eth mainnet? Do the chainanalysis contracts on other chains have the same data?

Reply:
None of them are very good but yes the Ethereum contract has the most addresses registered and afaict all the other addresses are also registered on the Ethereum contract.

A correct implementation would treat EVM addresses as equivalent, as risk has nothing to do with the chain.